### PR TITLE
feat: add manifest for shim

### PR DESCRIPTION
**Pull request by septs**

### CODE REVIEW

**Comment by Jack-Works** [comment]
Hold on, I need to review this.

File: .vscode/schema/manifest-shim.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
```
Comment (Jack-Works) [inline .vscode/schema/manifest-shim.json:1]: This file should be maintained in https://github.com/DimensionDev/webextension-polyfill as one of its public API.

File: .vscode/schema/manifest-shim.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    "$schema": "http://json-schema.org/draft-07/schema",
+    "type": "object",
+    "properties": {
```
Comment (Jack-Works) [inline .vscode/schema/manifest-shim.json:4]: Please allow extra properties

File: public/manifest-shim.json [hunk 3]
```diff
@@ -0,0 +1,4 @@
+{
+    "bypass-url-scheme-list": ["wc"],
```
Comment (Jack-Works) [inline public/manifest-shim.json:2]: It should end with `:`

File: public/manifest-shim.json [hunk 4]
```diff
@@ -0,0 +1,4 @@
...
+    "bypass-url-domain-list": ["arweave.net"]
```
Comment (Jack-Works) [inline public/manifest-shim.json:3]: maybe `external-uri` then 

```json
external-uri: [
    "wc:*" // Wallet Connect,
    "https?://*.arweave.net/*"
]
```

File: .vscode/schema/manifest-shim.json (hunk 2 above)
Comment (Jack-Works) [inline .vscode/schema/manifest-shim.json:4]: Now we have an ad-hoc `iOS-injected-scripts` in our iOS build. We may add it here as:

```typescript
"userscripts": string[]
```

**Review by Jack-Works — CHANGES_REQUESTED** [review]

File: public/manifest-shim.json (hunk 4 above)
Comment (septs) [reply public/manifest-shim.json:3]: iOS / Android not support JSON with Comment.

File: .vscode/settings.json [hunk 5]
```diff
@@ -25,6 +25,10 @@
         {
             "fileMatch": ["/packages/maskbook/src/_locales/*/messages.json"],
             "url": "/.vscode/schema/locale.json"
+        },
+        {
+            "fileMatch": ["/public/manifest-shim.json"],
+            "url": "/.vscode/schema/manifest-shim.json"
```
[the remark this answers is not in the document]
Comment (septs) [reply .vscode/settings.json:31]: no need

File: public/manifest.webextension-shim.json [hunk 6]
```diff
@@ -0,0 +1,4 @@
+{
+    "external-uri": ["^wc:", "^https?://arweave.net"],
+    "user-scripts": []
```
Comment (Jack-Works) [inline public/manifest.webextension-shim.json:3]:
```suggestion
    "user-scripts": ["js/injected-script.js"]
```

File: public/manifest.webextension-shim.json [hunk 7]
```diff
@@ -0,0 +1,4 @@
+{
+    "external-uri": ["^wc:", "^https?://arweave.net"],
```
Comment (Jack-Works) [inline public/manifest.webextension-shim.json:2]: Is it RegExp? If so, what standard of RegExp is it?

File: public/manifest.webextension-shim.json [hunk 8]
```diff
@@ -0,0 +1,4 @@
+{
```
Comment (Jack-Works) [inline public/manifest.webextension-shim.json:1]: Should it be a JSONc file? (JSON with comment)

**Review by Jack-Works — CHANGES_REQUESTED** [review]

File: .vscode/settings.json (hunk 5 above)
[the remark this answers is not in the document]
Comment (Jack-Works) [reply .vscode/settings.json]: By putting it in `.vscode/settings.json`, this file must be in this special position to be auto-completed. If you copy it anywhere else, it no longer works. This argument applies to the Chrome extension manifest as well.

File: .vscode/schema/manifest.webextension-shim.json [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+        },
+        "user-scripts": {
+            "type": "array",
+            "description": "user scripts file path list",
+            "items": {
+                "type": "string"
+            }
+        }
+    },
+    "required": ["external-uri", "user-scripts"]
```
Comment (Jack-Works) [inline .vscode/schema/manifest.webextension-shim.json:20]: They're all optional in API perspective

**Review by Jack-Works — CHANGES_REQUESTED** [review]

File: public/manifest.webextension-shim.json (hunk 7 above)
Comment (septs) [reply public/manifest.webextension-shim.json:2]: Perl Compatible?

File: public/manifest.webextension-shim.json (hunk 8 above)
Comment (septs) [reply public/manifest.webextension-shim.json:1]: for Android and iOS, need add new dependencies.

File: .vscode/schema/manifest.webextension-shim.json (hunk 9 above)
Comment (septs) [reply .vscode/schema/manifest.webextension-shim.json:20]: Why?

Comment (Jack-Works) [reply .vscode/schema/manifest.webextension-shim.json:20]: What if they don't have anything external? What if they don't have user-scripts?

**Comment by Jack-Works** [comment]
Let me move the schema to https://github.com/DimensionDev/webextension-polyfill

**Review by Jack-Works — APPROVED** [review]

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**